src/autopas/selectors/ContainerSelector.h

```diff
@@ -135,19 +141,21 @@ ContainerSelector<Particle, ParticleCell>::generateContainer(ContainerOptions co
       break;
     }
     default: {
-      AutoPasLog(warn, "Container type {} is not a known type!", utils::StringUtils::to_string(containerChoice));
+      utils::ExceptionHandler::exception("Container type {} is not a known type!",
+                                         utils::StringUtils::to_string(containerChoice));
     }
   }
 
   // copy particles so they do not get lost when container is switched
-  if (_optimalContainer != nullptr) {
-    for (auto particleIter = _optimalContainer->begin(IteratorBehavior::ownedOnly); particleIter.isValid();
+  if (_currentContainer != nullptr) {
+    for (auto particleIter = _currentContainer->begin(IteratorBehavior::haloAndOwned); particleIter.isValid();
          ++particleIter) {
-      container->addParticle(*particleIter);
-    }
-    for (auto particleIter = _optimalContainer->begin(IteratorBehavior::haloOnly); particleIter.isValid();
-         ++particleIter) {
-      container->addHaloParticle(*particleIter);
+      // try to add every particle as inner. If it fails try as a halo.
+      try {
+        container->addParticle(*particleIter);
+      } catch (const autopas::utils::ExceptionHandler::AutoPasException &e) {
+        container->addHaloParticle(*particleIter);
```

Comment:
what happens here, if exceptions are disabled?

Reply:
:boom: 
I actually do not expect AutoPas to work properly if exceptions are disabled